lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1583,7 +1583,30 @@ void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &source,
   // a local build with explicit modules and precise compiler
   // invocations isn't supported yet.
   if (cc1) {
-    dest.insert(dest.end(), source.begin(), source.end());
+    clang::CompilerInvocation invocation;
+    llvm::SmallVector<const char *> clangArgs;
+    clangArgs.reserve(source.size());
+    llvm::for_each(source, [&](const std::string &Arg) {
+      clangArgs.push_back(Arg.c_str());
+    });
+
+    clang::DiagnosticsEngine clangDiags(new clang::DiagnosticIDs(),
+                                        new clang::DiagnosticOptions(),
+                                        new clang::IgnoringDiagConsumer());
+    if (!clang::CompilerInvocation::CreateFromArgs(
+          invocation, clangArgs, clangDiags)) {
+      // FIXME: arguments failed to parse so the arguments from source are
+      // invalid. Don't add any arguments for now. Maybe log the error.
+      return;
```

Comment:
Can you just call `AddDiagnostic(eSeverityError, message)`?
Hopefully there's a DiagnosticConsumer that can easily convert to a string?

Reply:
I don't know what is a proper way to hook up (note this is inside a static method). The best way is probably hook up to the clang diagnostics handler and let it emit the error directly maybe? Currently I put in a placeholder ignoreDiagnosticConsumer.